File: native/Android/ProduceDatabase.java
package com.example.produceapp.data;

import androidx.room.Database;
import androidx.room.RoomDatabase;

// 新增功能：離線快取機制 (Offline Caching)
// Room Database 定義，負責管理本地端快取資料庫
@Database(entities = {ProduceEntity.class}, version = 1, exportSchema = false)
public abstract class ProduceDatabase extends RoomDatabase {
    public abstract ProduceDao produceDao();
}

File: native/Android/app/src/main/java/com/example/produceapp/data/ProduceDatabase.kt
package com.example.produceapp.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [ProduceEntity::class], version = 1, exportSchema = false)
abstract class ProduceDatabase : RoomDatabase() {
    abstract fun produceDao(): ProduceDao

    companion object {
        @Volatile
        private var INSTANCE: ProduceDatabase? = null

        fun getDatabase(context: Context): ProduceDatabase {
            return INSTANCE ?: synchronized(this) {
                val instance = Room.databaseBuilder(
                    context.applicationContext,
                    ProduceDatabase::class.java,
                    "produce_database"
                ).build()
                INSTANCE = instance
                instance
            }
        }
    }
}
